man/vignette_childs/_qgisprocess_basics.Rmd

```diff
@@ -0,0 +1,309 @@
+rp_tp = qgis_run_algorithm(
+  "sagang:addrastervaluestopoints",
+  SHAPES = random_points,
+  GRIDS = list(info$aspect[[1]], info$slope[[1]], info$tcurvature[[1]]),
+  RESAMPLING = 0)
+```
+
+To verify that it worked, read in the output.
+
+```{r multilayer-output}
+sf::st_as_sf(rp_tp)
+```
+# Piping
+
+`qgis_process` does not lend itself naturally to piping because its first argument is the name of a geoalgorithm instead of a data object.
+`qgis_pipe()` circumvents this by accepting a `.data` object as its first argument, and pipes this data object into the the first argument of a geoalgorithm assuming that the specified geoalgorithm needs a data input object as its first argument.
+
+```{r buf-pipe}
+system.file("longlake/longlake_depth.gpkg", package = "qgisprocess") |>
+  qgis_pipe("native:buffer", DISTANCE = 50)
+```
+
+If `.data` is a `qgis_result` object, `qgis_pipe()` automatically tries to select an element named `OUTPUT`.
+However, if the output has another name (e.g., `DEM_PREPROC` as in the example below) or if there are multiple output elements to choose from (e.g., `sagang:sagawetnessindex` has four output rasters, check with `qgis_outputs("sagang:sagawetnessindex")`), you can specify the wanted output object via the `.select` argument.
+Please note that we make sure that temporary output raster files, i.e., all output rasters we do not specifically name ourselves, should use SAGA's native raster file format by setting the `qgisprocess.tmp_raster_ext` to `.sdat`.
+Using the default raster output format `.tif` might lead to trouble depending on the installed versions of third-party packages (GDAL, SAGA, etc.). 
+
+```{r qgis-pipe}
+dem <- system.file("raster/dem.tif", package = "spDataLarge")
+# in case you need to enable the SAGA next generation algorithms, run the following lines:
+# qgis_enable_plugins("processing_saga_nextgen")
+
+options(qgisprocess.tmp_raster_ext = ".sdat")
+qgis_run_algorithm(algorithm = "sagang:sinkremoval", DEM = dem,
+                   METHOD = 1) |>
+  qgis_pipe("sagang:sagawetnessindex", .select = "DEM_PREPROC")
+```
+
+When piping, `qgis_pipe()` automatically cleans up after you by deleting intermediate results.
+This avoids cluttering your system when running geoalgorithms on large spatial data files.
+To turn off this behavior, set `.clean` to `FALSE`.
+
+Of course, you can also pipe to `qgis_run_algorithm()` by manually extracting the `OUTPUT` object and redirecting it to the appropriate input argument of the next processing step.
+This avoids ambiguity and allows for greater flexibility though it might not be as convenient as `qgis_pipe()`.
+For example, intermediate results remain on disk for the duration of your R session.
+
+```{r native pipe}
+out = qgis_run_algorithm(algorithm = "sagang:sinkremoval", DEM = dem, 
+                         METHOD = 1) |>
+  _$DEM_PREPROC[1] |>
+  qgis_run_algorithm(algorithm = "sagang:sagawetnessindex",
+                     DEM = _)
```

Comment:
I had expected that below approach:

```r
 |>
  qgis_run_algorithm(algorithm = "sagang:sagawetnessindex",
                     DEM = _$DEM_PREPROC[1])
```

would work too, but unfortunately it yields an error 'invalid use of pipe placeholder'.

This may reflect the experimental nature of this feature in R 4.3.0, where it is announced as:

> As an experimental feature the placeholder _ can now also be used in the rhs of a forward pipe |> expression as the first argument in an extraction call, such as _$coef. More generally, it can be used as the head of a chain of extractions, such as _$coef[[2]].

Reply:
yes, I expected the same :-)

Reply:
Let's keep it as-is, but will need to be simplified when this placeholder performs better in future. Currently the anonymous function with lambda syntax even looks better :wink:.